llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -10261,7 +10261,6 @@ SDValue RISCVTargetLowering::lowerINSERT_SUBVECTOR(SDValue Op,
   // and decomposeSubvectorInsertExtractToSubRegs takes this into account. So if
   // we have a fixed length subvector, we need to adjust the index by 1/vscale.
   if (SubVecVT.isFixedLengthVector()) {
-    assert(VLen);
     unsigned Vscale = *VLen / RISCV::RVVBitsPerBlock;
```

Comment:
Is * on an empty optional not UB?

Reply:
I think [llvm optional used to](https://github.com/llvm-mirror/llvm/blob/2c4ca6832fa6b306ee6a7010bfb80a3f2596f824/include/llvm/ADT/Optional.h#L73) std does not.

> The behavior is undefined if *this does not contain a value.
https://en.cppreference.com/w/cpp/utility/optional/operator*